.travis.yml

```diff
@@ -1,31 +1,95 @@
+language: perl
+
 branches:
   except:
     - /^issue\d+/
     - /^gh\d+/
-language: perl
-matrix:
-   fast_finish: true
-#   allow_failures:
-#     - perl: "5.12"
-#     - perl: "5.10"
+
 env:
   global:
     - PERL_USE_UNSAFE_INC=0
     - AUTHOR_TESTING=1
     - AUTOMATED_TESTING=1
     - RELEASE_TESTING=1
+
 perl:
-  - "5.28"
-  - "5.26"
-  - "5.24"
-  - "5.22"
-  - "5.20"
-  - "5.18"
-  - "5.16"
-  - "5.14"
+  - "5.6.0"
+  - "5.6.0-thr"
+  - "5.6.1"
+  - "5.6.1-thr"
+  - "5.6.2"
+  - "5.6.2-thr"
+  - "5.8.1"
+  - "5.8.1-thr"
+  - "5.8.2"
+  - "5.8.2-thr"
+  - "5.8.3"
+  - "5.8.3-thr"
+  - "5.8.4"
+  - "5.8.4-thr"
+  - "5.8.5"
+  - "5.8.5-thr"
+  - "5.8.6"
+  - "5.8.6-thr"
+  - "5.8.7"
+  - "5.8.7-thr"
+  - "5.8.8"
+  - "5.8.8-thr"
+  - "5.8.9"
+  - "5.8.9-thr"
+  - "5.8.0"
+  - "5.8.0-thr"
+  - "5.10.0"
+  - "5.10.0-thr"
+  - "5.10.1"
+  - "5.10.1-thr"
+  - "5.10-thr"
+  - "5.12.0"
+  - "5.12.0-thr"
   - "5.12"
-  - "5.10"
-  - "5.8"
-script:
-  - perl Makefile.PL && make test
+  - "5.12-thr"
+  - "5.14"
+  - "5.14-thr"
+  - "5.16"
+  - "5.16-thr"
+  - "5.18"
+  - "5.18-shrplib"
+  - "5.20"
+  - "5.20-shrplib"
+  - "5.22"
+  - "5.22-shrplib"
+  - "5.24"
+  - "5.24-shrplib"
+  - "5.26"
+  - "5.26-thr"
+  - "5.28"
+  - "5.28-thr"
+  - "5.30"
+  - "5.30-thr"
+  - "dev"
+  - "dev-thr"
+  - "blead"
+  - "blead-thr"
 
+matrix:
+#   fast_finish: true
+#   allow_failures:
+#     - perl: "5.12"
+#     - perl: "5.10"
+  include:
+    - perl: system_precise
+      dist: precise
+    - perl: system_trusty
+      dist: trusty
+
+before_install:
+  - if ! [[ $TRAVIS_PERL_VERSION =~ ^system ]] && [ -z "$PERLBREW_PERL" ]; then eval $(curl https://travis-perl.github.io/init) --perl; fi
+
+install:
+  - perl Makefile.PL
+  - make regen
```

Comment:
You can add:
```yaml
  - git diff --quiet
```
after `make regen` to ensure that all files are in git up-to-date. make regen may change some files if they are not up-to-date. And `git diff --quiet` returns non-zero status code if some files in git are changed.